### DISCUSSION TITLE
Emergency Access

### DISCUSSION THREAD

**Discussion by steviehs** [dani-garcia/vaultwarden#1987] [Q&A] [answered]
Uh, I am so excited to test Emergency Access... but how to access it? I see the Emergency schedules in the Admin interface, but nothing appears in WebVault... Did I something wrong while installing/updating? WebVault shows 2.21.1
Diagnostics tells me the same, 2.21.1 installed and there is an "update" to 2.23.0 but how to update?

If somebody could give me a hint, I will add the documentation in the wiki :-)

**Comment by BlackDex** [comment, marked as answer]
Have some patients :).
The web-vault v2.23.0 has just been build like 2 hours ago.
We need to trigger a new build for vaultwarden to have it included within the image.

If you can't wait, you can download the tar.gz from here https://github.com/dani-garcia/bw_web_builds/releases/tag/v2.23.0.

Extract that to your `/vw-data` folder into a new directory called `web` or something. And configure `WEB_VAULT_FOLDER` to point to `/data/web`.

**Comment by steviehs** [comment]
Haha, of course I was impatience. 
That looks great! A first test with manual granting worked flawlessly. Now I will check the scheduled granting.
Thanks for all your work!

**Reply by snevas** [reply]
Just wondering as I'm looking through the new feature: the wait time is bypassable for anybody who can set the system time of the hosting machine to the future, right?
This would mean you would be effectively handing over your vault to someone when he/she is the invitee and can also set the host's time.

**Reply by BlackDex** [reply]
Well, you invited that person for a reason i think, you should also receive an e-mail regarding this so you would be notified in the event that would happen.

But if someone has access to your server i think you also have some other issues.
Also, if someone really wants and have access to your server, then they could also change the wait time in the database.
Or, the can even fake the time for vaultwarden it self without changing the whole system/host time.

**Reply by snevas** [reply]
True, so that might be something worth considering when using this feature in a self-hosting situation. When it's hosted at Bitwarden, you know for sure that someone will only get access after the wait period as they don't have access to these resources you've described (division of trust).
For complete unnoticeable access to someones vault (if a sysadmin gets invited as a emergency contact) you could restore the db to a separate environment without mail set-up, request access and set the time forward or change the wait time in db.

I can't think of a technical way of enforcing the wait time (cryptographically) in a self-hosted scenario though.

**Reply by BlackDex** [reply]
Well, in your case i wouldn't self host anything because you seem to can't trust the system.

**Reply by bokkabonga** [reply]
I agree with BlackDex. If someone has root access to your host you have a lot of other problems at hand.

**Reply by snevas** [reply]
Totally agree with that statement as well, but also loved the fact that vaultwarden passwords could never be decrypted unless you had the master password of a specific vault (thanks to the clientside encrypting/decrypting). So imho it's kind of a system where you don't need complete trust of the environment you are hosting your vault in. This just adds a little sidenote to it when leaving emergency access enabled e.g., that the wait time is bypassable and it can be used unnoticeable by the grantor if the grantee has access to a backup.
Thanks for this discussion guys!
